ui/views/sync.jsx

```diff
@@ -28,6 +33,212 @@ function isNotLAN (peer) {
+    this.state.renderer.dispose()
+  }
+
+  setupRenderer () {
+    const el = ReactDOM.findDOMNode(this)
+    const renderer = createRenderer(this.state.graph, el)
+
+    renderer.run()
+
+    this.setState({
+      renderer: renderer
+    })
+  }
+
+  render () {
+    return <div className="peer-graph-container" />
+  }
+}
+
+function createRenderer (graph, el) {
+  let renderer = ngraphSvg(graph, {
+    container: el,
+    // defaults
+    //physics: {
+    //  springLength: 30,
+    //  springCoeff: 0.0008,
+    //  dragCoeff: 0.01,
+    //  gravity: -1.2,
+    //  theta: 1
+    //}
+    physics: {
+      springLength: 20,
+      springCoeff: 0.0001,
+      dragCoeff: 0.01,
+      gravity: -1.5,
+      theta: 0.7
+      //theta: 0.8
+    }
+  }).node( function nodeBuilder(node) {
+    const isFriend = social.follows(app.user.id, node.data.id) && social.follows(node.data.id, app.user.id)
+    const isRelayPeer = node.data.isRelayPeer
+
+    let radius = 5
+    let color = '#fff'
+    const strokeColor = '#f00'
+    const strokeWidth = 1
+
+    if (node.data.isUser) { 
+      radius = 15
+      color = '#f00'
+    }
+    if (isFriend) {
+      color = '#f00'
+    }
+    if (isRelayPeer) radius = 10
+    if (node.data.isConnected) color = 'green' 
+
+    return ngraphSvg.svg("circle", {
+      r: radius,
+      cx: 2*radius,
+      cy: 2*radius,
+      fill: color,
+      stroke: strokeColor,
+      'stroke-width': strokeWidth,
+
+      text: node.data ? node.data.name : undefined
+    })
+  }).placeNode( function nodePositionCallback(nodeUI, pos) {
+    nodeUI.attr("cx", pos.x).attr("cy", pos.y)
+  }).link( function linkBuilder(linkUI, pos) {
+    const fromNode = graph.getNode(linkUI.fromId)
+    const toNode = graph.getNode(linkUI.toId)
+
+    //link is from or to a 'pub'
+    const involvesRelayPeer = fromNode.data.isRelayPeer || toNode.data.isRelayPeer 
+    const isLinkingUser = linkUI.toId === app.user.id || linkUI.fromId === app.user.id
+
+    let linkOpacity = 0
+    let linkStrokeWidth = 1
+    if (isLinkingUser && involvesRelayPeer) {
+      linkOpacity = 0.6
+      linkStrokeWidth = 5
+    } else if (involvesRelayPeer) {
+      linkOpacity = 0.15
+      linkStrokeWidth = 3
+    }
+
+    //const linkStroke = involvesRelayPeer ? '#F00' : 'grey'
+    //const linkStrokeDasharray = involvesRelayPeer ? 1 : '5, 5'
+    const linkStroke = '#F00'
+    const linkStrokeDasharray = isLinkingUser ? '1,1' : '1, 2'
+
+    return ngraphSvg.svg("line", {
+      stroke: linkStroke,
+      'stroke-width': linkStrokeWidth,
+      'stroke-dasharray': linkStrokeDasharray,
+      opacity: linkOpacity
+    })
+  })
+
+  let group = ngraphSvg.svg("g", {transform: "translate(10 20)"})
+  let text = ngraphSvg.svg("text", {fill: 'black'}).text("John Dot Awesome") 
+  let img = ngraphSvg.svg("image", {
+    //'xlink:href': 'http://localhost:7777/&xISx10wRtEirc1+LEHp0khJc5lrsybHBnIry78pan9U=.sha256?fallback=img&name=JohnDotAwesome.png',
```

Comment:
@pfraze I'm having a heaps of trouble getting an image to render in SVG inside Patchwork.

Note the same pattern works in JSFiddle http://jsfiddle.net/4s6mucuq/
(assuming blob names are the same .... )

you can see on link 169 / 170 I've tried changing these attributes and this comes through in electron, but then displays nothing ... 

if you uncomment this line you get an error about invalid namespace. Not sure what's needed there but have tried a few things. ngraphSvg.svg is actually just this file, and setting attributes uses this function which is called setAttributeNS  https://github.com/anvaka/simplesvg/blob/master/index.js#L81

---

if you could spin it up and confirm I'm not crazy or missing something dumb like z-index that would be rad.
Further to that, any suggestions ?